### PR TITLE
cron_ticket_email_parser.php - send new ticket opened email to unknown clients

### PR DESCRIPTION
**Pull request by cybershieldms**
- Will now send unknown client (client_id = 0) a ticket open notification email 
- Added regex to not parse bounce e-mails (e.g. From: postmaster, mailer-daemon)
- Fixed ticket description missing from new ticket notifications sent to others.
- Added E-Mail From line display name to beginning of ticket description/e-mail quote with e-mail address and date.

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://ticketfeatures1076.pr-review.itflow.org  
 _(automatic message)_

File: cron_ticket_email_parser.php [hunk 1]
```diff
@@ -475,9 +477,11 @@ function getInboxFolder($client, $inboxNames) {
                     }
                 } elseif ($config_ticket_email_parse_unknown_senders)  {
                     // Parse even if the sender is unknown
-
-                    if (addTicket(0, 'Guest', $from_email, 0, $date, $subject, $message_body, $message->getAttachments(), $original_message_file)) {
-                        $email_processed = true;
+		    $bad_from_pattern = "/daemon|postmaster|reply|root|admin/i";
```
Comment (wrongecho) [inline cron_ticket_email_parser.php:480]: I like this concept, but would probably omit 'reply' and 'admin'. 

We probably want automated emails from things like noreply@microsoft.com to be parsed

Some clients use admin@ for shared mailboxes.

**Comment by wrongecho** [comment]
Hey @cybershieldms 

Thanks for the PR! :)

I've tested and this seems to work nicely. I like the way you've put the persons name in bold and email in brackets too!

image

Great concept with excluding bounces but I'd suggest we exclude admin/reply as these can often be used legitimately/for messages you want.

File: cron_ticket_email_parser.php (hunk 1 above)
Comment (cybershieldms) [reply cron_ticket_email_parser.php:480]: I agree.  I could see actually making the block list regex a configurable field later on in Settings.  Might take a stab at that later on.  I'll fix it now to just exclude daemon and postmaster.

**Comment by cybershieldms** [comment]
> Great concept with excluding bounces but I'd suggest we exclude admin/reply as these can often be used legitimately/for messages you want.

@wrongecho - Thanks!  First time with Github.   Done... Changed the regex to just include postmaster and daemon.

In the future, I could see a configurable variable in Settings to make a customizable block list.

**Comment by wrongecho** [comment]
> @wrongecho - Thanks!  First time with Github.   Done... Changed the regex to just include postmaster and daemon.
> 
> In the future, I could see a configurable variable in Settings to make a customizable block list.

You've done great for your first time!! :)

Agreed, it would be nice to make the ignore list configurable.

**Review by wrongecho — APPROVED** [review]

**Comment by johnnyq** [comment]
awesome, we'll pull this one in

**Comment by wrongecho** [comment]
Nice work Craig!! :)

**Pull request merged**